delivery_line_sale_line_position/report/report_deliveryslip.xml

```diff
@@ -26,18 +23,28 @@
                 t-value="any(o.move_line_ids.mapped('position_sale_line'))"
             />
         </xpath>
-        <xpath expr="//th[@name='th_sml_product']" position="before">
-            <th t-if="has_line_position" name="th_sml_pos"><strong>Pos</strong></th>
+        <xpath
+            expr="//table[@name='stock_move_line_table']/thead//th[1]"
+            position="before"
+        >
+            <th t-if="has_line_position and has_serial_number" name="th_sml_pos"><strong
+                >Pos</strong></th>
         </xpath>
+        <!--
+            NOTE: The rows are rendered by other templates.
+            `stock_report_delivery_has_serial_move_line` if the lines have lots/serials
+            `stock_report_delivery_aggregated_move_lines` if they don't. In this case the lines are
+            aggregated, and so we don't display the position.
+        -->
     </template>
 
     <template
         id="stock_report_delivery_has_serial_move_line"
         inherit_id="stock.stock_report_delivery_has_serial_move_line"
     >
-        <xpath expr="//span[@t-field='move_line.product_id']/.." position="before">
+        <xpath expr="//td[1]" position="before">
```

Comment:
the former selector seems more precise than this

Reply:
Yes but the point is not to be anchored to the move_line.product_id, but rather to the first td (whatever it is).
That is, we don't care what the existing columns are, we want ours to be the first.